src/cluster/multi_step_cluster.cpp

```diff
@@ -146,30 +156,17 @@ vector<int> MultiStep::cluster_sets(const size_t nb_size, vector<TempFile*> &sor
 	return cluster;
 }
 
-unordered_map<uint32_t, NodEdgSet> MultiStep::find_connected_components(vector<uint32_t>& sindex, const vector <size_t>& nedges){
-	vector<uint32_t> seen_nodes;
+unordered_map<uint32_t, NodEdgSet> MultiStep::find_connected_components(const vector<unordered_set<uint32_t>> &connected, vector<uint32_t> &EdgSet, const vector <size_t>& nedges){
+	
 	unordered_map<uint32_t, NodEdgSet> ne;
-
-	uint32_t curr = 0;
-	for (size_t k = 0; k < sindex.size(); k++) {
-		curr = sindex[k];
-		while (sindex[curr] != curr) {
-			seen_nodes.push_back(sindex[curr]);
-			curr = sindex[curr];
-		}
-
-		sindex[k] = curr;
-		for (size_t i : seen_nodes) {
-			sindex[i] = curr;
+	for (size_t i = 0; i < connected.size(); i++) {
+		for(size_t j = 0; j < connected[i].size(); j++){
```

Comment:
Damit die Semantik stimmt, solltest du hier über den Inhalt des `unordered_set` iterieren, also `for(uint32_t j : connected[i])`.